source/agora/consensus/data/ConsensusParams.d

```diff
@@ -28,19 +30,34 @@ public immutable class ConsensusParams
     /// Maximum number of nodes to include in an autogenerated quorum set
     public uint MaxQuorumNodes;
 
+    /// The Genesis block of the chain
+    public Block Genesis;
+
     /***************************************************************************
 
         Constructor
 
         Params:
+            genesis = Genesis block to use for this chain
             validator_cycle = cycle length for a validator
             max_quorum_nodes = max nodes to include in autogenerated quorum set
 
     ***************************************************************************/
 
-    public this (uint validator_cycle = 1008, uint max_quorum_nodes = 7)
+    public this (immutable(Block) genesis, uint validator_cycle = 1008,
+                 uint max_quorum_nodes = 7)
     {
+        this.Genesis = genesis;
         this.ValidatorCycle = validator_cycle;
         this.MaxQuorumNodes = max_quorum_nodes;
     }
+
+    /// Transitional method to avoid having to rewrite all unittests
+    /// TODO: Remove / replace with something more principled
+    version (unittest) public this (
+        uint validator_cycle = 1008, uint max_quorum_nodes = 7)
+    {
+        import agora.consensus.data.genesis.Test;
+        this(GenesisBlock, validator_cycle);
```

Comment:
I guess it's kind of complicated to check in even this class to confirm which genesis block is used for the Genesis block. But, I'm not sure what's best.

Reply:
I don't follow your question ? The `GenesisBlock` symbol refers to the one in `agora.consensus.data.genesis.Test`.
I could turn it into a selective import to make it more obvious, but I was hoping the scoped symbol would do that.

Reply:
Fixed

Reply:
I know what you mean, but I just thought it would be good to set the genesis block at higher level classes which are using the ConsensusParams class. Anyway, thanks for making the code more understandable.

Reply:
I originally did that, but the diff was huge, and very repetitive. Most of the time, you want to use the test genesis block in `unittest`.

Reply:
I think it was a good decision if there were so many changes and repetitive code.